// @rocket.chat/apps-engine/server/permissions/checkers/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.permissionCheckers = void 0;
const AppActivationBridge_1 = require("./AppActivationBridge");
const AppApisBridge_1 = require("./AppApisBridge");
const AppCloudBridge_1 = require("./AppCloudBridge");
const AppCommandsBridge_1 = require("./AppCommandsBridge");
const AppDetailChangesBridge_1 = require("./AppDetailChangesBridge");
const AppEnvironmentalVariableBridge_1 = require("./AppEnvironmentalVariableBridge");
const AppHttpBridge_1 = require("./AppHttpBridge");
const AppInternalBridge_1 = require("./AppInternalBridge");
const AppLivechatBridge_1 = require("./AppLivechatBridge");
const AppMessageBridge_1 = require("./AppMessageBridge");
const AppPersistenceBridge_1 = require("./AppPersistenceBridge");
const AppRoomBridge_1 = require("./AppRoomBridge");
const AppSchedulerBridge_1 = require("./AppSchedulerBridge");
const AppSettingBridge_1 = require("./AppSettingBridge");
const AppUiInteractionBridge_1 = require("./AppUiInteractionBridge");
const AppUploadBridge_1 = require("./AppUploadBridge");
const AppUserBridge_1 = require("./AppUserBridge");
exports.permissionCheckers = {
    AppActivationBridge: AppActivationBridge_1.AppActivationBridge,
    AppApisBridge: AppApisBridge_1.AppApisBridge,
    AppCommandsBridge: AppCommandsBridge_1.AppCommandsBridge,
    AppCloudBridge: AppCloudBridge_1.AppCloudBridge,
    AppDetailChangesBridge: AppDetailChangesBridge_1.AppDetailChangesBridge,
    AppEnvironmentalVariableBridge: AppEnvironmentalVariableBridge_1.AppEnvironmentalVariableBridge,
    AppHttpBridge: AppHttpBridge_1.AppHttpBridge,
    AppInternalBridge: AppInternalBridge_1.AppInternalBridge,
    AppLivechatBridge: AppLivechatBridge_1.AppLivechatBridge,
    AppMessageBridge: AppMessageBridge_1.AppMessageBridge,
    AppPersistenceBridge: AppPersistenceBridge_1.AppPersistenceBridge,
    AppRoomBridge: AppRoomBridge_1.AppRoomBridge,
    AppSchedulerBridge: AppSchedulerBridge_1.AppSchedulerBridge,
    AppSettingBridge: AppSettingBridge_1.AppSettingBridge,
    UiInteractionBridge: AppUiInteractionBridge_1.AppUiInteractionBridge,
    AppUploadBridge: AppUploadBridge_1.AppUploadBridge,
    AppUserBridge: AppUserBridge_1.AppUserBridge,
};
